internal/charmstore/store.go

```diff
@@ -227,6 +227,17 @@ func (p *Pool) requestStoreNB(always bool) (*Store, error) {
 	return store, nil
 }
 
+// newBlobstore creates a new blobstore based on config and the given StoreDatabase.
+func (p *Pool) newBlobstore(db *mgo.Database) *blobstore.Store {
+	// If no BlobStoreProviders are in the config, use the legacy config of
+	// same mongo as entity database and entitystore as the gridfs prefix.
+	if 0 == len(p.config.BlobStorageProviders) {
+		logger.Debugf("using default gridfs blobstore with prefix entitystore - no provider-config")
+		return blobstore.New(db, "entitystore")
+	}
+	return blobstore.NewMultiStore(p.config.BlobStorageProviders)
```

Comment:
This is pretty heavyweight to be doing on every charmstore request. Better to change the interface so we don't need to create the multistore and parse all the configuration every time. We should chat about this - there are a bunch of possibilities.